### ISSUE TITLE
Can't build autofpga

### ISSUE THREAD

**Issue by fjpolo** [ZipCPU/autofpga#5]
Hi, I'm trying to build autofpga from a docker container in Ubuntu22.04 and I'm getting the following error:

```
# make
make --no-print-directory --directory=sw  
bash runme.sh
Opened autofpga.dbg
ERR: Could not open wbdown.txt
Searched through .:../auto-data
ERR: PARSE-ERR: Could not open wbdown.txt 
          Searched through: .:../auto-data
WARNING: Too many interrupts assigned to PIC altpic
Clock: hdmi_out_clk, is i_hdmi_out_clk at 148500 kHz        
preset != NULL
Clock: clk, is i_clk at 100 MHz, w/ associated reset i_reset
Clock: net_rx_clk, is i_net_rx_clk at 125 MHz
Clock: hdmi_in_clk, is i_hdmi_in_clk at 148500 kHz
Clock: hdmi_in_hsclk, is i_hdmi_in_hsclk at 1485 MHz        
All clocks enumerated
Bus: wbflash
Bus: rambus
FATAL ERR: Bus wbwide has no defined clock
exit code  1
```
There's some stuff in `sw/`:
```
# ls -lh sw/
total 4.3M
-rw-r--r-- 1 root root 4.1K Oct 28 16:19 Makefile
-rw-r--r-- 1 root root 5.2K Oct 28 16:19 ast.cpp 
-rw-r--r-- 1 root root 3.8K Oct 28 16:19 ast.h
-rwxr-xr-x 1 root root 3.0M Oct 28 16:19 autofpga
-rw-r--r-- 1 root root  52K Oct 28 16:19 autofpga.cpp
-rw-r--r-- 1 root root 816K Oct 28 16:29 autofpga.dbg
-rw-r--r-- 1 root root 1.6K Oct 28 16:19 automdata.h
-rw-r--r-- 1 root root 1.9K Oct 28 16:19 autopdata.h
-rw-r--r-- 1 root root 2.0K Oct 28 16:19 bitlib.cpp
-rw-r--r-- 1 root root 1.7K Oct 28 16:19 bitlib.h
-rw-r--r-- 1 root root  11K Oct 28 16:19 bldboardld.cpp
-rw-r--r-- 1 root root 1.6K Oct 28 16:19 bldboardld.h
-rw-r--r-- 1 root root 5.5K Oct 28 16:19 bldcachable.cpp
-rw-r--r-- 1 root root 1.6K Oct 28 16:19 bldcachable.h
-rw-r--r-- 1 root root  14K Oct 28 16:19 bldregdefs.cpp
-rw-r--r-- 1 root root 2.7K Oct 28 16:19 bldregdefs.h
-rw-r--r-- 1 root root 5.3K Oct 28 16:19 bldrtlmake.cpp
-rw-r--r-- 1 root root 1.8K Oct 28 16:19 bldrtlmake.h
-rw-r--r-- 1 root root  15K Oct 28 16:19 bldsim.cpp
-rw-r--r-- 1 root root 2.0K Oct 28 16:19 bldsim.h
-rw-r--r-- 1 root root  12K Oct 28 16:19 bldtestb.cpp
-rw-r--r-- 1 root root 1.8K Oct 28 16:19 bldtestb.h
drwxr-xr-x 2 root root 4.0K Oct 28 16:19 bus
-rw-r--r-- 1 root root  33K Oct 28 16:19 businfo.cpp
-rw-r--r-- 1 root root 5.4K Oct 28 16:19 businfo.h
-rw-r--r-- 1 root root  15K Oct 28 16:19 clockinfo.cpp
-rw-r--r-- 1 root root 2.9K Oct 28 16:19 clockinfo.h
-rw-r--r-- 1 root root 3.2K Oct 28 16:19 expr.l
-rw-r--r-- 1 root root  51K Oct 28 16:19 expr.tab.cpp
-rw-r--r-- 1 root root 3.8K Oct 28 16:19 expr.tab.h
-rw-r--r-- 1 root root 4.5K Oct 28 16:19 expr.ypp
-rw-r--r-- 1 root root 3.6K Oct 28 16:19 gather.cpp
-rw-r--r-- 1 root root 1.8K Oct 28 16:19 gather.h
-rw-r--r-- 1 root root 5.3K Oct 28 16:19 genbus.cpp
-rw-r--r-- 1 root root 4.4K Oct 28 16:19 genbus.h
-rw-r--r-- 1 root root 1.5K Oct 28 16:19 globals.cpp
-rw-r--r-- 1 root root 1.6K Oct 28 16:19 globals.h
-rw-r--r-- 1 root root 8.7K Oct 28 16:19 ifdefs.cpp
-rw-r--r-- 1 root root 2.1K Oct 28 16:19 ifdefs.h
-rw-r--r-- 1 root root 8.8K Oct 28 16:19 keys.cpp
-rw-r--r-- 1 root root 6.6K Oct 28 16:19 keys.h
-rw-r--r-- 1 root root  12K Oct 28 16:19 kveval.cpp
-rw-r--r-- 1 root root 2.2K Oct 28 16:19 kveval.h
-rw-r--r-- 1 root root 4.7K Oct 28 16:19 legalnotice.cpp
-rw-r--r-- 1 root root 2.0K Oct 28 16:19 legalnotice.h
-rw-r--r-- 1 root root  50K Oct 28 16:19 lex.yy.cpp
-rw-r--r-- 1 root root  11K Oct 28 16:19 lex.yy.h
-rw-r--r-- 1 root root  28K Oct 28 16:19 mapdhash.cpp
-rw-r--r-- 1 root root 4.1K Oct 28 16:19 mapdhash.h
-rw-r--r-- 1 root root 2.4K Oct 28 16:19 mlist.cpp
-rw-r--r-- 1 root root 2.0K Oct 28 16:19 mlist.h
-rw-r--r-- 1 root root 3.1K Oct 28 16:19 msgs.cpp
-rw-r--r-- 1 root root 2.2K Oct 28 16:19 msgs.h
drwxr-xr-x 2 root root 4.0K Oct 28 16:19 obj-pc
-rw-r--r-- 1 root root 8.2K Oct 28 16:19 parser.cpp
-rw-r--r-- 1 root root 2.1K Oct 28 16:19 parser.h
-rw-r--r-- 1 root root  17K Oct 28 16:19 plist.cpp
-rw-r--r-- 1 root root 4.9K Oct 28 16:19 plist.h
-rw-r--r-- 1 root root 6.2K Oct 28 16:19 predicates.cpp
-rw-r--r-- 1 root root 3.2K Oct 28 16:19 predicates.h
-rw-r--r-- 1 root root  397 Oct 28 16:29 runme.sh
-rw-r--r-- 1 root root 3.8K Oct 28 16:19 subbus.cpp
-rw-r--r-- 1 root root 1.9K Oct 28 16:19 subbus.h
```
But running make in `autofpga-demo` leads to lots of errors:
```
wbregs.cpp:165:33: note: 'printf' is defined in header '<cstdio>'; did you forget to '#include <cstdio>'?
wbregs.cpp:167:25: error: 'printf' was not declared in this scope
  167 |                         printf("%08x (%8s) : [%c%c%c%c] %08x\n", address, nm,
      |                         ^~~~~~
wbregs.cpp:167:25: note: 'printf' is defined in header '<cstdio>'; did you forget to '#include <cstdio>'?
wbregs.cpp:171:25: error: 'printf' was not declared in this scope
  171 |                         printf("%08x (%8s) : BUS-ERROR\n", address, nm);
      |                         ^~~~~~
wbregs.cpp:171:25: note: 'printf' is defined in header '<cstdio>'; did you forget to '#include <cstdio>'?
wbregs.cpp:176:17: error: 'printf' was not declared in this scope
  176 |                 printf("%08x (%8s)-> %08x\n", address, nm, value);
      |                 ^~~~~~
wbregs.cpp:176:17: note: 'printf' is defined in header '<cstdio>'; did you forget to '#include <cstdio>'?
wbregs.cpp:180:17: error: 'printf' was not declared in this scope
  180 |                 printf("FPGA was interrupted\n");
      |                 ^~~~~~
wbregs.cpp:180:17: note: 'printf' is defined in header '<cstdio>'; did you forget to '#include <cstdio>'?
make[1]: *** [Makefile:67: obj-pc/wbregs.o] Error 1
make: *** [Makefile:145: sw] Error 2
```

**Comment by ZipCPU** [comment]
It looks like AutoFPGA, the program, built just fine.  The demo appears to not be building due to a missing file.  That fix should be simple enough (just provide the missing file).  I can fix that.

In the meantime, you should have no problem using the tool.

**Comment by ZipCPU** [comment]
Also, between now and when I get the autofpga-demo project back up and running again (it probably needs a bit of love, I haven't touched it in a while), recent projects which have used AutoFPGA include [kimos](https://github.com/ZipCPU/kimos), [eth10g (a 10Gb Ethernet switch)](https://github.com/ZipCPU/eth10g), and the [VideoZip upgrade](https://github.com/ZipCPU/videozip/tree/dev).  Feel free to check these projects out.  In each case, `make autodata` from the main directory will run AutoFPGA to build the various project components.

**Comment by murattokez** [comment]
hi,

I had the same problem. How can I solve it. I cannot compile autofpga in any way.

**Comment by ZipCPU** [comment]
This should be fixed as of fdfd1b633e9ccd79e279012d8dfd341d9c4cd8a5
